remotes/resolver.go

```diff
@@ -65,7 +65,7 @@ type FetcherByDigest interface {
 	// FetcherByDigest usually returns an incomplete descriptor.
 	// Typically, the media type is always set to "application/octet-stream",
 	// and the annotations are unset.
-	FetchByDigest(ctx context.Context, dgst digest.Digest) (io.ReadCloser, ocispec.Descriptor, error)
+	FetchByDigest(ctx context.Context, desc digest.Digest, mediatype string) (io.ReadCloser, ocispec.Descriptor, error)
```

Comment:
Why not just use `Fetcher.Fetch()` ?

Reply:
there are cases when we only have mediatype and digest but do not have a full descriptor struct (size might be unknown)